lib/indexer.rb

```diff
@@ -0,0 +1,63 @@
+module Argo
+  class Indexer
+    ##
+    # Returns a Logger instance for recording info about indexing attempts
+    # @yield attempt to execute 'entry_id_block' and use the result as an extra identifier for the log 
+    #   entry.  a placeholder will be used otherwise. 'request.uuid' might be useful in a Rails app.
+    def self.generate_index_logger(&entry_id_block)
+      index_logger = Logger.new(Settings.INDEXER.LOG, Settings.INDEXER.LOG_ROTATION_INTERVAL)
+      index_logger.formatter = proc do |severity, datetime, progname, msg|
+        date_format_str = Argo::Config.date_format_str
+        entry_id = begin entry_id_block.call rescue '---' end
```

Comment:
i'm open to better ways of trying a block to get a result, and falling back to a default if the block doesn't return something usable for any reason.  but what hound's complaining about seems ok to me.

is the objection one of style (not the most readable/maintainable way)?  or one of robustness (running whatever and just rescuing is fragile in some way i can't see)?